lib/surface/compiler.ex

```diff
@@ -430,6 +435,15 @@ defmodule Surface.Compiler do
     end)
   end
 
+  defp has_attribute?([], _), do: false
+
+  defp has_attribute?(attributes, attr_name) do
+    Enum.find_value(attributes, false, fn
+      {^attr_name, _, _} -> true
+      _ -> nil
```

Comment:
Since this function ends with `?`, it's probably a good idea to guarantee it returns a boolean.

```suggestion
      _ -> false
```

Reply:
It always returns a boolean due to the `false` argument in `find_value`. Replacing `nil` by `false` and remove the second attribute of `find_value` will return `nil`. Replacing `nil` by `false` and keeping `false` as a default value will return `false` because of the default value and not because of `_ -> false`. This being said, it gains in clarity.

Reply:
You're absolutely right. I missed the default value. Sorry. Just leave as it is or, if you want to make it more explicit, you can use:

```elixir
Enum.any?(attributes, &match?({^attr_name, _, _}, &1))
```

It's less overhead in my mind as reading it describes exactly what it does but I think we're stepping on the field of personal taste now.  So just ignore me, if you want. 🙂

Reply:
Thanks for your proposition! I The intent is clearer than mine without the ambiguity I introduced ;)